be/src/olap/rowset/rowset_meta.cpp

```diff
@@ -102,18 +102,20 @@ void RowsetMeta::set_fs(io::FileSystemSPtr fs) {
     _fs = std::move(fs);
 }
 
-void RowsetMeta::to_rowset_pb(RowsetMetaPB* rs_meta_pb) const {
+void RowsetMeta::to_rowset_pb(RowsetMetaPB* rs_meta_pb, bool skip_schema) const {
     *rs_meta_pb = _rowset_meta_pb;
-    if (_schema) {
-        _schema->to_schema_pb(rs_meta_pb->mutable_tablet_schema());
+    if (_schema) [[likely]] {
+        rs_meta_pb->set_schema_version(_schema->schema_version());
+        if (!skip_schema) {
+            // For cloud, separate tablet schema from rowset meta to reduce persistent size.
+            _schema->to_schema_pb(rs_meta_pb->mutable_tablet_schema());
+        }
     }
 }
 
-RowsetMetaPB RowsetMeta::get_rowset_pb() {
+RowsetMetaPB RowsetMeta::get_rowset_pb(bool skip_schema) const {
     RowsetMetaPB rowset_meta_pb = _rowset_meta_pb;
```

Comment:
seems unnecessary to assign init.
```suggestion
    RowsetMetaPB rowset_meta_pb;
```